components/ca-file-input.js

```diff
@@ -0,0 +1,253 @@
+define(['/components/helpers/create-element.js', '/components/helpers/dialogs.js', 'document-register-element'], (createElement, dialogs) => {
+    const mimeTypeToExtension = {
+        'application/pdf': { accept: '.pdf', label: 'PDF' },
+        'image/png': { accept: '.png', label: 'PNG' },
+        'image/jpeg': { accept: '.jpg,.jpeg', label: 'JPG' },
+        'image/gif': { accept: '.gif', label: 'GIF' },
+        'application/msword': { accept: '.doc,.dot', label: 'DOC' },
+        'application/vnd.ms-excel': { accept: '.xls,.xlm,.xla,.xlc,.xlt,.xlw', label: 'XLS' }
+    };
+
+    const txtProp = ('textContent' in document.createElement('i')) ? 'textContent' : 'innerText';
+
+    /**
+     * FileInput class, creates instance of ca-file-input web component
+     */
+    class FileInput extends HTMLElement {
+        /**
+         * Called when the component is created (yet to be attached to the DOM).
+         * @returns {undefined} void fuction that scaffolds based structure and binds events.
+         */
+        createdCallback() {
+            const self = this;
```

Comment:
Look very carefully at the inner event listeniner, notice the use of this. Refactor without breaking.